auth/src/main/java/org/aerogear/mobile/security/SecurityService.java

```diff
@@ -65,7 +65,7 @@ public AsyncSecurityCheckExecutor getAsyncCheckExecutor() {
      * @throws IllegalArgumentException if {@param securityCheckType} is null
      */
     public SecurityCheckResult check(@NonNull final SecurityCheckType securityCheckType) {
-        return (SecurityCheckResult) nonNull(securityCheckType, "securityCheckType").getSecurityCheck();
+        return check(nonNull(securityCheckType, "securityCheckType").getSecurityCheck());
```

Comment:
I don't think this works.
This should be 
`return check(nonNull(securityCheckType, securityCheckType").getSecurityCheck()).test(context);`

Reply:
@ziccardi Why call test here instead of providing a SecurityCheck to the other check method which calls test anyway? Are we not duplicating the test call if we do this?

Reply:
@aidenkeating @ciaranRoche Forget my comment. I misread the code. Sorry.